registry/auth.go

```diff
@@ -74,6 +74,12 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 	return "", nil
 }
 
+// Checks that requests to the v2 registry can be authorized.
+func (auth *RequestAuthorization) IsAuthorized() bool {
```

Comment:
Happy to make adjustments, but I'm not sure I understand the feedback...

It sounds like you want this fallback to be able to occur on 403 to the registry server, but you don't want any partial data to be written to the registry server.

We could get some additional coverage by making "IsAuthorized" require a "200 OK" from "/v2/".  This would partially cover the case of sending "Basic" or "Bearer" to the registry.

However, I don't see a way that we can benignly verify that the appropriate scopes have been included by the auth endpoint.  In particular, what will verify the "push" scope without uploading data?